### PR TITLE
HERITAGE-210 Integrate the Map view into the search results page

### PR DESCRIPTION
**Pull request by garethmead**
Ticket URL: [HERITAGE-210](https://national-archives.atlassian.net/browse/HERITAGE-210)

## About these changes

Adds map prototype with hard coded data

## How to check these changes

Map can be viewed at `/search/catalogue/?group=community&views=map`

## Before assigning to reviewer, please make sure you have

- [ ] Checked things thoroughly before handing over to reviewer
- [ ] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [ ] Ensured that PR includes only commits relevant to the ticket
- [ ] Waited for all CI jobs to pass before requesting a review
- [ ] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

[HERITAGE-210]: https://national-archives.atlassian.net/browse/HERITAGE-210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/search/blocks/search_results_map.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="ohos-search-results-map-container">
+    <div id="ohos-search-results-map" class="ohos-search-results-map" data-test="{{ default_geo_data }}" data-js-default-lat="{{ default_geo_data.lat }}" data-js-default-lon="{{ default_geo_data.lon }}" data-js-default-zoom="{{ default_geo_data.zoom }}"></div>
+    <div class="ohos-search-results-map__info">
+        <i class="ohos-search-results-map__info-icon fa-solid fa-circle-info"></i>
+        <div>
+            <p>Map view shows only records with location data. <a href="#">View all search results.</a></p>
```
Comment (TNA-Allan) [inline templates/search/blocks/search_results_map.html:6]: `<a href="#">View all search results.</a>`  href is empty over here. Is this in scope?

File: templates/search/blocks/visualisation_menu.html [hunk 2]
```diff
@@ -1,16 +1,18 @@
+{% load search_tags %}
+
 <!--visualisation secondary navigation-->
 <div class="tna-container">
     <div class="tna-column tna-column--full">
         <div class="search-visualisation-nav">
             <ul class="search-visualisation-nav__list" data-id="search-visualisation-nav-list">
                 <li class="search-visualisation-nav__list-item">
-                    <a href="/" class="search-visualisation-nav__link" aria-current="true">
+                    <a href="?{% query_string_include 'views' views.LIST.value %}" class="search-visualisation-nav__link"{% if form.views.value == views.LIST.value %} aria-current="true"{% endif %}>
```
Comment (TNA-Allan) [inline templates/search/blocks/visualisation_menu.html:9]: 1. Is List Map view in scope for this ticket?
2. What is the expected behaviour/url link of List View, when a Search criteria are applied on a page and the user then navigates to Map view?

File: templates/search/catalogue_search.html [hunk 3]
```diff
@@ -31,48 +43,63 @@
...
 
 {% block extra_js %}
-    <script src="{% static 'scripts/catalogue_search.js' %}"></script>
+    {% if form.views.value != views.MAP.value %}
+        <script src="{% static 'scripts/catalogue_search.js' %}"></script>
+    {% endif %}
+
+    {% if form.views.value == views.MAP.value %}
+        <script src="{% static 'scripts/search_results_map.js' %}"></script>
+    {% endif %}
```
Comment (TNA-Allan) [inline templates/search/catalogue_search.html:104]:
```suggestion
    {% if form.views.value == views.MAP.value %}
        <script src="{% static 'scripts/search_results_map.js' %}"></script>
    {% else %}
        <script src="{% static 'scripts/catalogue_search.js' %}"></script>
    {% endif %}
```
Why not use an else here?

File: templates/search/catalogue_search.html [hunk 4]
```diff
@@ -31,48 +43,63 @@
             {% include './blocks/visualisation_menu.html' %}
         {% endif %}
 
-        {% if buckets.current.result_count > 0 %}
+        {# Map view is displayed without results and filters #}
+        {% if form.views.value == views.MAP.value %}
             <div class="tna-container">
                 <div class="tna-column tna-column--full">
-                    {% include './blocks/search_sort_and_view_options.html' %}
+                    {% include './blocks/search_results_map.html' %}
```
Comment (TNA-Allan) [inline templates/search/catalogue_search.html:50]: Why are the div tna-container, tna-column tna-column--full - outside of block html - search_results_map.html? Is there any issue to include then inside the block html?

File: templates/search/catalogue_search.html [hunk 5]
```diff
@@ -31,48 +43,63 @@
...
-        {% endif %}
-
-        {% if buckets.current.result_count > 0 or form.errors %}
-            <div class="tna-container tna-!--margin-top-s" data-id="catalogue-search-grid">
-                <div class="tna-column tna-column--width-1-3 tna-column--full-medium tna-column--full-small tna-column--full-tiny tna-!--margin-top-s" data-id="catalogue-search-sidebar">
-                    {% include './blocks/search_filters.html' %}
+        {% else %}
+            {% if buckets.current.result_count > 0 %}
+                <div class="tna-container">
+                    <div class="tna-column tna-column--full">
```
Comment (TNA-Allan) [inline templates/search/catalogue_search.html:56]: same as above - Why are the div tna-container, tna-column tna-column--full - outside of block html? Is there any issue to include then inside the block html?

**Review by TNA-Allan — CHANGES_REQUESTED** [review]

**Comment by JohnHeeryTNA** [comment]
Have pulled the latest changes and runs well locally but a couple of issues found.

1. http://127.0.0.1:8000/static/images/icons/**icon-map-pin.svg** is not found, may need adding into the repo.

2. With long descriptions we are getting some overflow issues (see screenshot), would suggest adding a max-width rule, some thing like :

```
.leaflet-popup-content p {
  max-width:350px;
}
```
Screenshot 2024-02-23 at 10 50 55

File: templates/search/catalogue_search.html (hunk 4 above)
Comment (ahosgood) [reply templates/search/catalogue_search.html:50]: I would suggest leaving them outside like this. That means that you could then include your block in whatever layout is needed, rather than the block defining the layout.

Comment (TNA-Allan) [reply templates/search/catalogue_search.html:50]: Thanks for that @ahosgood , came to know in a discussion that its used this way in other places as well. 
On another note, lesser number of `div`'s make it easier to navigate to the what we're looking for. Is there any limit or practice for nesting div's?

File: templates/search/blocks/visualisation_menu.html (hunk 2 above)
Comment (TNA-Allan) [reply templates/search/blocks/visualisation_menu.html:9]: Discussed in stand up, a separate ticket will be created to address the navigation behaviour

**Review by TNA-Allan — APPROVED** [review]
Looks good. Thanks @garethmead

**Review by JohnHeeryTNA — APPROVED** [review]
Approving from FE perspective.
